### PR TITLE
feat: adds refactoring section

### PR DESCRIPTION
**Pull request by brofori**

### CODE REVIEW

**Review by dpommeranz — APPROVED** [review]

File: book/refactoring.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Refactoring
```
Comment (reasn) [inline book/refactoring.md:1]: The article is not linked anywhere

File: book/refactoring.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+### Guide to Cleaning up the Codebase
+
+> Stay disciplined when refactoring when following the approaches described in this article
```
Comment (reasn) [inline book/refactoring.md:5]: All other articles use different bullet points

File: book/refactoring.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+> Encourage others to write better code (comment in PR, in pair-programming sessions, etc.)
+> ...
```
Comment (reasn) [inline book/refactoring.md:7]: ...

File: book/refactoring.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+> Long term code profit
+
+* [www.sitepoint.com/write-maintainable-javascript](https://www.sitepoint.com/write-maintainable-javascript/)
```
Comment (reasn) [inline book/refactoring.md:10]: External links should come with an explanation

**Review by reasn — CHANGES_REQUESTED** [review]

**Comment by reasn** [comment]
Partly covered by https://guidelines.stylelounge.io/book/refactoring.html
Removed the external link because it's unrelated to the process of refactoring.

**Pull request closed without merging**